src/hints/execution.rs

```diff
@@ -1871,4 +1887,29 @@ mod tests {
             get_integer_from_var_name(vars::ids::STATE_ENTRY, &mut vm, &ids_data, &ap_tracking).unwrap().into_owned();
         assert_eq!(state_entry, Felt252::from(123));
     }
+
+    #[test]
+    fn test_set_fp_plus_4_to_tx_nonce() {
+        let mut vm = VirtualMachine::new(false);
+        vm.add_memory_segment();
+        vm.add_memory_segment();
+        vm.set_fp(3);
+
+        let ap_tracking = ApTracking::new();
+        let constants = HashMap::new();
+
+        let ids_data = HashMap::new();
+
+        // insert tx with a nonce
+        let mut tx = InternalTransaction::default();
+        tx.nonce = Some(Felt252::THREE);
+        let mut exec_scopes: ExecutionScopes = Default::default();
+        exec_scopes.insert_value(vars::scopes::TX, tx);
+
+        set_fp_plus_4_to_tx_nonce(&mut vm, &mut exec_scopes, &ids_data, &ap_tracking, &constants).unwrap();
+
+        let address: Relocatable = (vm.get_fp() + 4usize).unwrap();
+        let value = vm.get_integer(address).unwrap();
+        assert_eq!(value, std::borrow::Cow::Borrowed(&Felt252::THREE));
```

Comment:
Isn't it simpler to just `assert_eq!(*value, Felt252::THREE)`? Or deref one line above?

Reply:
Good point, cleaned this up in 45bc33e